packages/abstract-core/src/objects/namespace.rs

```diff
@@ -0,0 +1,125 @@
+use cosmwasm_std::StdResult;
+use cw_storage_plus::{Key, KeyDeserialize, Prefixer, PrimaryKey};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use std::fmt::Display;
+
+use crate::AbstractResult;
+
+use super::module::validate_name;
+
+#[derive(Deserialize, Serialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Namespace(String);
+
+impl Namespace {
+    pub fn new(namespace: &str) -> Self {
+        Self(namespace.to_owned())
+    }
+    pub fn as_str(&self) -> &str {
+        &self.0
+    }
+    pub fn validate(&self) -> AbstractResult<()> {
+        validate_name(&self.0)?;
+        Ok(())
+    }
+}
+
+impl From<&str> for Namespace {
+    fn from(namespace: &str) -> Self {
+        Self::new(namespace)
+    }
+}
+
+impl From<String> for Namespace {
+    fn from(namespace: String) -> Self {
+        Self::new(&namespace)
+    }
+}
+
+impl From<&String> for Namespace {
```

Comment:
small comment: you can practically always use &str instead of &String (https://rust-unofficial.github.io/patterns/idioms/coercion-arguments.html)

Reply:
👍